File: 30 days of code/Day 09 - Recursion/Recursion.cs
/**
 *              RECURSION | C#
 * Domain       30 DAYS OF CODE
 * Sub domain   DAY 09
 */

using System;
using System.IO;

class Program
{
    static void Main(string[] args)
    {
        var reader = new InputReader(false);
        int n = reader.NextInt;
        reader.Close();

        int result = Solve(n);
        Console.WriteLine(result);
    }

    static int Solve(int n)
    {
        if (n == 0)
        {
            return 1;
        }

        return n * Solve(n - 1);
    }
}

class InputReader
{
    private TextReader input;

    public InputReader(bool fromFile)
    {
        input = fromFile ? new StreamReader("input.txt") : Console.In;
    }

    public string NextLine => input.ReadLine();

    public int NextInt => int.Parse(NextLine);

    public void Close() => input.Close();
}

File: 30 days of code/Day 09 - Recursion/Recursion.cpp
/**
*              RECURSION | C++
* Domain       30 DAYS OF CODE
* Sub domain   DAY 09
*/

#include "stdafx.h"
#include <iostream>
#include <fstream>

using namespace std;

ifstream file("input.txt");
istream& f = file;

int factorial(int n) {
	if (n == 0) {
		return 1;
	}
	return n * factorial(n - 1);
}

void main() {
	int n; f >> n;

	int result = factorial(n);
	cout << result << endl;
}